trl/trainer/ppo_trainer.py

```diff
@@ -305,21 +308,19 @@ def __init__(
             self.lr_scheduler,
         )
         if is_deepspeed_used:
-            # 8 bit models are already set on the correct device
+            # Quantized models are already set on the correct device
             if not self.is_peft_model and not (
                 getattr(self.ref_model.pretrained_model, "is_loaded_in_8bit", False)
                 or getattr(self.ref_model.pretrained_model, "is_loaded_in_4bit", False)
             ):
-                # DS integration only allows for single model and as `ref_model` is only used for
-                # `KL divergence loss`,i.e, in eval model, just have it be on the respective device and
-                # there is no need to pass it to the `accelerator.prepare` call
-                self.ref_model = self.ref_model.to(self.accelerator.device)
-
-            # this hack seems to be needed for DS stage 3 to work
-            if self.accelerator.state.deepspeed_plugin.zero_stage == 3:
-                self.model.train()
```

Comment:
This has been moved to the training loop where I think it should be done for all models (including DeepSpeed ones)